crates/accelerate/src/remove_diagonal_gates_before_measure.rs

```diff
@@ -0,0 +1,96 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+/// Remove diagonal gates (including diagonal 2Q gates) before a measurement.
+use pyo3::prelude::*;
+use std::collections::HashSet;
+
+use qiskit_circuit::dag_circuit::{DAGCircuit, NodeType};
+use qiskit_circuit::operations::Operation;
+use qiskit_circuit::operations::StandardGate;
+
+/// Run the RemoveDiagonalGatesBeforeMeasure pass on `dag`.
+/// Args:
+///     dag (DAGCircuit): the DAG to be optimized.
+/// Returns:
+///     DAGCircuit: the optimized DAG.
+#[pyfunction]
+#[pyo3(name = "remove_diagonal_gates_before_measure")]
+fn run_remove_diagonal_before_measure(_py: Python, dag: &mut DAGCircuit) -> PyResult<()> {
```

Comment:
Oh, the other thing is, if we're not interacting with Python at all we don't need to have a py token argument. This makes the function easier to call from rust if we need to.

```suggestion
fn run_remove_diagonal_before_measure(dag: &mut DAGCircuit) -> PyResult<()> {
```